src/editors/data/services/cms/api.js

```diff
@@ -29,6 +29,35 @@ export const apiMethods = {
       data,
     );
   },
+  deleteTranscript: ({
+    studioEndpointUrl,
+    language,
+    blockId,
+    videoId,
+  }) => {
+    const deleteJSON = { data: { lang: language, edx_video_id: videoId } };
+    return deleteObject(
+      urls.videoTranscripts({ studioEndpointUrl, blockId }),
+      deleteJSON,
+    );
+  },
+  uploadTranscript: ({
+    blockId,
+    studioEndpointUrl,
+    transcript,
+    videoId,
+    language,
+  }) => {
+    const data = new FormData();
+    data.append('file', transcript);
+    data.append('edx_video_id', videoId);
+    data.append('language_code', language);
+    data.append('new_language_code', language);
+    return post(
+      urls.videoTranscripts({ studioEndpointUrl, blockId }),
+      data,
+    );
```

Comment:
Can you upload a transcript without an edx_video_id?

Reply:
I don't think so.